accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,188 @@
+-----------------------
+
+Current situation: an image is built on Buildhost, then pushed to SUMA Server htdocs directory. Then SUMA creates a db entry (including package list) and a pillar file.
+The pillar conains URL of image tarball, checksums and additional deployment details. It is used for image synchronization and deployment. These data are extracted 
+on Buildhost at the end of build phase. Some of the data can't be extracted later form the tarball.
+
+The SUMA API related to images is rather docker-centered. For Kiwi images it allows to list images and packages and delete an image.
+
+Use cases currently not covered:
+
+- Move image tarball to external HTTP server
+  - In large environments it makes sense to use one or more servers for distribution of images, it should not be the Hub itself.
+- Copy image metadata from one SUMA Server to another 
+  - Hub - copy to all servers
+  - Test in lab, then move the same image to production.
+- Adjust deployment details
+
+
+Proposed solution:
+
+- Implement API calls to
+  - Get Kiwi Image pillar
+  - Update Kiwi Image pillar
+  - Export Kiwi Image (complete metadata)
+  - Import Kiwi Image
+
+- Implement a tool that handles the use cases above
+
+
+HWTYPE configurations
+---------------------
+
+This is currently implemented as a formula assigned to a group with special name.
+
+Proposed solution:
+Extend retail_yaml to create the groups and formula data via Hub and Peripheral Server API.
+
+Alternative:
+Central creation of grups is covered in [Hub Users RFC](00061-hub-users.md). The salt states
+can be extended to handle also formula data.
+
+
+Combined SUMA and Branch server
+-------------------------------
+
+Branch services are configured by Salt formulas. Also PXE entries for each terminal are generated
+by Salt (currently via reactor).
+With combined SUMA and Branch server, these formulas should be applied on SUMA server itself.
+
+Alternative 1: Hub as a Salt Master for Combined SUMA and Branch server
+
+- this is already used for [Hub Users RFC](00061-hub-users.md)
+
+problems:
+- does not work for PXE entries and disconnected operation
+- mixed environment with Hub, Peripheral Servers with standalone Branch Serveres and 
+  Combined Peripheral Servers will have non-optimal configuration
+- does not cover the "Single server" setup
+
+Alternative 2: Multi-master setup of Combined SUMA and Branch server
+
+- Salt minion running on Combined SUMA and Branch server has 2 masters - Hub and itself
+
+problems:
+- SUMA registered to itself cause many problems, this can't be supported
+
+
+Alternative 3: Use salt-call --local for Branch server part of combined server
+
+- how to represent such setup in GUI and API ?
+- would need significant changes in SUMA code
+
+
+Alternative 4: Virtualize Branch server part of combined server
+
+- use separate VM / container for the Branch server
+- the minion connects to SUMA on the same physical host
+
+- solves all the problems of other alternatives
```

Comment:
What needs to be done to support this alternative? Is it just documentation?
Can we prepare OS Image template for branch server image and use virtualization support of Uyuni/SUSE Manager to deploy it on itself? With that comes question as some users run Uyuni in virtualized environment, nested virtualization will need to be allowed.

Reply:
I don't know yet.

Reply:
Containerizing the Proxy is something that was imagined and even attempted in the past. Solving this case might be an additional good side-effect, although I am not yet sold on throwing in containerization yet.

Reply:
Proxy is not needed, terminals can connect to Peripheral servers directly.

This is about containerizing salt minion and the Branch services.